src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -796,68 +548,36 @@ private void InterceptException(Action action)
         /// Sets a new writer state; verifies that the transition from the current state into new state is valid.
         /// </summary>
         /// <param name="newState">The writer state to transition into.</param>
-        private void SetState(BatchWriterState newState)
-        {
-            this.InterceptException(() => this.ValidateTransition(newState));

```

Comment:
SetState should remain private -- all state management should be done in the base class.  Logic that is manipulating state in the derived classes needs to be moved back to the base class.